src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -987,22 +1014,122 @@ private static ConnectionMultiplexer CreateMultiplexer(object configuration, Log
         /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
         public static ConnectionMultiplexer Connect(string configuration, TextWriter log = null)
         {
-            SocketConnection.AssertDependencies();
-            return ConnectImpl(configuration, log);
+            return Connect(PrepareConfig(configuration), log);
         }
 
         /// <summary>
         /// Create a new ConnectionMultiplexer instance
         /// </summary>
-        /// <param name="configuration">The configurtion options to use for this multiplexer.</param>
+        /// <param name="configuration">The configuration options to use for this multiplexer.</param>
         /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
         public static ConnectionMultiplexer Connect(ConfigurationOptions configuration, TextWriter log = null)
         {
             SocketConnection.AssertDependencies();
-            return ConnectImpl(configuration, log);
+
+            bool sentinel = !string.IsNullOrEmpty(configuration?.ServiceName);
```

Comment:
Thought; private method this to DRY up? e.g.:
```cs
if (IsSentinel(configuration))
```

Reply:
Done